src/base/name/label.rs

```diff
@@ -479,12 +482,12 @@ impl OwnedLabel {
         let mut res = [0u8; 64];
         while let Some(ch) = chars.next() {
             if res[0] as usize >= Label::MAX_LEN {
-                return Err(LabelFromStrError::LongLabel);
+                return Err(LabelFromStrErrorEnum::LongLabel.into());
             }
             let ch = match ch {
                 ' '..='-' | '/'..='[' | ']'..='~' => ch as u8,
                 '\\' => parse_escape(&mut chars, res[0] > 0)?,
-                _ => return Err(LabelFromStrError::IllegalCharacter(ch)),
+                _ => return Err(BadSymbol::non_ascii().into()),
```

Comment:
This new approach seems to lose context. Is it not useful to pass the offending character out?